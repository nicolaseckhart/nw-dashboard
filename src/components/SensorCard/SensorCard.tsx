import * as React from 'react';
import { Sensor, SensorMeasurement } from '../../shared';
import { Col, Row } from 'react-bootstrap';
import { Link } from 'react-router-dom';

interface Props {
  sensor: Sensor;
}

export const SensorCard: React.FC<Props> = ({ sensor }: Props) => {
  const historyLink = `/history/${sensor.identifier}/day`;

  const renderSensorMeasurement = (measurement: SensorMeasurement, key: number) => (
    <Row key={key}>
      <Col xs={{ span: 1 }}>
        <span className={`jam jam-${measurement.icon}`} />
      </Col>
      {measurement.name && <Col xs={{ span: 7 }}>{measurement.name}</Col>}
      <Col xs={{ span: 2 }} className="text-right text-nowrap">
        {measurement.value}
      </Col>
      <Col xs={{ span: 1 }} className="pl-0">
        {measurement.unit}
      </Col>
    </Row>
  );

  return (
    <Col md={{ span: 4 }} className="my-3">
      <Link to={historyLink}>
        <div className={`info-card ${sensor.accent}-accent`}>
          <h4 className="text-muted font-weight-lighter">{sensor.name}</h4>
          {sensor.data.map((m: SensorMeasurement, i: number) => renderSensorMeasurement(m, i))}
        </div>
      </Link>
    </Col>
  );
};
