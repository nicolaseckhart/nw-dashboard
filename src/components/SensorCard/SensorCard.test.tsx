import * as React from 'react';
import { SensorCard } from './SensorCard';
import { shallow, ShallowWrapper } from 'enzyme';
import * as factories from '../../../test/factories';

function environmentSensorCard(): ShallowWrapper {
  const environmentSensor = factories.getSensorData().sensors[0];
  return shallow(<SensorCard sensor={environmentSensor} />);
}

describe('SensorCard', () => {
  it('renders the component correctly', () => {
    expect(environmentSensorCard()).toMatchSnapshot();
  });
});
