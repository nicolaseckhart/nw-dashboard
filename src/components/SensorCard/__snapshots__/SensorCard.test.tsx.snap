// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SensorCard renders the component correctly 1`] = `
<Col
  className="my-3"
  md={
    Object {
      "span": 4,
    }
  }
>
  <Link
    to="/history/intakecoolant/day"
  >
    <div
      className="info-card blue-accent"
    >
      <h4
        className="text-muted font-weight-lighter"
      >
        Intake Coolant
      </h4>
      <Row
        key="0"
        noGutters={false}
      >
        <Col
          xs={
            Object {
              "span": 1,
            }
          }
        >
          <span
            className="jam jam-temperature"
          />
        </Col>
        <Col
          xs={
            Object {
              "span": 7,
            }
          }
        >
          Temperature
        </Col>
        <Col
          className="text-right"
          xs={
            Object {
              "span": 2,
            }
          }
        >
          14.9
        </Col>
        <Col
          className="pl-0"
          xs={
            Object {
              "span": 1,
            }
          }
        >
           Â°C
        </Col>
      </Row>
      <Row
        key="1"
        noGutters={false}
      >
        <Col
          xs={
            Object {
              "span": 1,
            }
          }
        >
          <span
            className="jam jam-cloud"
          />
        </Col>
        <Col
          xs={
            Object {
              "span": 7,
            }
          }
        >
          Humidity
        </Col>
        <Col
          className="text-right"
          xs={
            Object {
              "span": 2,
            }
          }
        >
          50.4
        </Col>
        <Col
          className="pl-0"
          xs={
            Object {
              "span": 1,
            }
          }
        >
          %
        </Col>
      </Row>
    </div>
  </Link>
</Col>
`;
